pkg/await/apps_deployment.go

```diff
@@ -460,6 +459,11 @@ func (dia *deploymentInitAwaiter) processDeploymentEvent(event watch.Event) {
 		}
 	}
 
+	// Since we can't tell for sure from this version of the API, mark ReplicaSet as available.
```

Comment:
I'm fine with this as a bug fix, but we should be able to actually account for this by looking at the pods themselves, right?

Reply:
Yeah, I think so. I'm inclined to rewrite this awaiter sooner rather than later because I'm having trouble wrapping my head around all of the nuances here.

Reply:
Still working on a fix for the `extensions/v1beta1` apiVersion, but pushing a fix for the other apiVersions in the meantime. I'll address the workaround separately